tests/integration_tests.rs

```diff
@@ -759,6 +767,113 @@ fn simple_for_dictmerger_loop() {
     unsafe { weld_value_free(ret_value) };
 }
 
+fn simple_groupmerger() {
+    #[allow(dead_code)]
+    struct Args {
+        x: WeldVec<i32>,
+        y: WeldVec<i32>,
+    }
+
+    let code = "|x:vec[i32], y:vec[i32]| tovec(result(for(zip(x,y), groupmerger[i32,i32], \
+                |b,i,e| merge(b, e))))";
+
+    let conf = default_conf();
+    let keys = [1, 2, 2, 3, 3, 1];
+    let vals = [2, 3, 4, 1, 0, 2];
+    let ref input_data = Args {
+        x: WeldVec {
+            data: &keys as *const i32,
+            len: keys.len() as i64,
+        },
+        y: WeldVec {
+            data: &vals as *const i32,
+            len: vals.len() as i64,
+        },
+    };
+
+    let ret_value = compile_and_run(code, conf, input_data);
+    let data = unsafe { weld_value_data(ret_value) as *const WeldVec<Pair<i32, WeldVec<i32>>> };
+    let result = unsafe { (*data).clone() };
+    let output : Vec<(i32, Vec<i32>)> = vec![(1, vec![2,2]), (2, vec![3,4]), (3, vec![1,0])];
+
+    let mut res : Vec<(i32, Vec<i32>)> = (0..result.len)
+        .into_iter()
+        .map(|x| {
+            let key = unsafe { (*result.data.offset(x as isize)).ele1 };
+            let val = unsafe { ((*result.data.offset(x as isize)).ele2).clone() };
+            let vec : Vec<i32> = (0..val.len)
+                .into_iter()
+                .map(|y| unsafe { *val.data.offset(y as isize) })
+                .collect();
+            (key, vec)
+        })
+        .collect();
+    res.sort_by_key(|a| a.0);
+
+    assert_eq!(res, output);
+    unsafe { weld_value_free(ret_value) };
+}
+
```

Comment:
@mateiz 
How about doing it this way?
It might be useful to have a function converting weld types to rust types.
I tried going down that route, but rust wouldn't allow me to dereference result.data.offset(x as isize)
in a separate function. It's okay with closures though (since it's in the same scope).